packages/contentstack-audit/src/modules/entries.ts

```diff
@@ -619,12 +641,155 @@ export default class Entries {
             entry[uid] as EntryGroupFieldDataType[],
           ) as EntryGroupFieldDataType;
           break;
+        case 'text':
+        case 'number':
+          if (field.hasOwnProperty('display_type')) {
+            entry[uid] = this.fixSelectField(
+              [...tree, { uid: field.uid, name: field.display_name, data_type: field.data_type }],
+              field as SelectFeildStruct,
+              entry[uid] as EntrySelectFeildDataType,
+            ) as EntrySelectFeildDataType;
+          }
+          break;
       }
     });
 
     return entry;
   }
 
+  /**
+   * We check for the select field with condition in order of multiple -> Array
+   * We find the missing values i.e. the values present in entry but not in the options of the content-type
+   * @param tree : Contains all the tree where the select field is located used for getting the path to it
+   * @param fieldStructure it contains the Content-type structure of the field
+   * @param field It contains the value that is present in the entry it can be array or value of number | string
+   * @returns if there is missing field returns field and path
+   * Else empty array
+   */
+  validateSelectField(tree: Record<string, unknown>[], fieldStructure: SelectFeildStruct, field: any) {
+    const { display_name, enum: selectOptions, multiple, min_instance, display_type } = fieldStructure;
+
+    let missingValues;
+
+    if (multiple) {
+      if (Array.isArray(field)) {
```

Comment:
@cs-raj do we need to mention else condition with logger?

Reply:
No as we only have two choices